Modules/Blueprints/src/selxBlueprintImpl.h

```diff
@@ -71,17 +75,18 @@ class BlueprintImpl
   // and holds component connection configuration settings
   struct ConnectionPropertyType
   {
-    ConnectionPropertyType( ParameterMapType parameterMap = {} ) :  parameterMap( parameterMap ) {}
+    ConnectionPropertyType( ConnectionNameType name = "", ParameterMapType parameterMap = {} ) :name( name ),  parameterMap( parameterMap ) {}
+    ConnectionNameType name;
     ParameterMapType parameterMap;
   };
 
   typedef boost::labeled_graph<
     boost::adjacency_list<
-    boost::vecS,
-    boost::vecS,
-    boost::bidirectionalS,
-    ComponentPropertyType,
-    ConnectionPropertyType
+    boost::multisetS, //OutEdgeList: allow multiple parallel connections
+    boost::vecS, // VertexList, 
+    boost::bidirectionalS, //Directed: directed graph, but give access to traverse in both directions
```

Comment:
What happens with the update order if we have a cyclic graph?

Reply:
You are right, I forgot to handle that case in this PR. Currently, we don't have cyclic graphs, but I plan to work on the CompositeTransformComponent for multistage registration were that will happen. Is it ok with you if I address handling cyclic graphs there? (I have a cyclic graph check solution in mind that addresses other expected update issues with CompositeTransformComponent as well)

Reply:
Yes that is fine :)

Reply:
#154